libnss/src/interop.rs

```diff
@@ -1,24 +1,72 @@
-use libc::c_int;
 use std::collections::VecDeque;
 use std::ffi::CString;
+use std::io;
+
+pub trait ToC<C> {
+    unsafe fn to_c(&self, result: *mut C, buffer: &mut CBuffer) -> std::io::Result<()>;
+}
 
 #[allow(dead_code)]
 pub enum NssStatus {
+    TryAgain = -2,
+    Unavail = -1,
+    NotFound = 0,
+    Success = 1,
+    Return = 2,
+}
+
+pub enum Response<R> {
     TryAgain,
     Unavail,
     NotFound,
-    Success,
+    Success(R),
     Return,
```

Comment:
You might wonder why a new enum which mostly mirrors `NssStatus`. I started by making `NssStatus` be generic over its result type, but I think separating these is important:

Hooks may want to have more granular failure types which will allow us to also map `errnop` (or even `h_errnop` in case of host hooks). So, I expect they look like the same enum right now but are in fact meaningfully different.